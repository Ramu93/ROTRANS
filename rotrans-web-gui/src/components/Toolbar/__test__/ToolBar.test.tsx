import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import validators from "../../../fixtures/validators";
import ToolBar from "../";

const mockStore = configureStore([]);

describe("ToolBar component", () => {
  let store;

  afterEach(cleanup);

  it("Rendered with validators and without selected validator", () => {
    store = mockStore({
      coreReducer: {
        validators: validators,
      },
    });

    const { getByTestId } = render(
      <Provider store={store}>
        <ToolBar title="Transactions" showValidatorsList />
      </Provider>
    );
    expect(getByTestId("toolbarTitle")).toHaveTextContent("Transactions");
    expect(getByTestId("toolbarSelectedAgentName")).toHaveTextContent(
      validators[0].agentName
    );
    expect(getByTestId("toolbarSelectedAgentIp")).toHaveTextContent(
      validators[0].agentIp
    );
  });

  it("Rendered with selected validator", () => {
    store = mockStore({
      coreReducer: {
        validators: validators,
        rootValidator: validators[1],
      },
    });

    const { getByTestId } = render(
      <Provider store={store}>
        <ToolBar title="Transactions" showValidatorsList />
      </Provider>
    );
    expect(getByTestId("toolbarSelectedAgentName")).toHaveTextContent(
      validators[1].agentName
    );
    expect(getByTestId("toolbarSelectedAgentIp")).toHaveTextContent(
      validators[1].agentIp
    );
  });

  it("Matches with snapshot - without selected validator", () => {
    store = mockStore({
      coreReducer: {
        validators: validators,
      },
    });

    const tree = renderer
      .create(
        <Provider store={store}>
          <ToolBar title="Visualization" showValidatorsList />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Matches with snapshot - without selected validator", () => {
    store = mockStore({
      coreReducer: {
        validators: validators,
        rootValidator: validators[1],
      },
    });

    const tree = renderer
      .create(
        <Provider store={store}>
          <ToolBar title="Transactions" showValidatorsList />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
