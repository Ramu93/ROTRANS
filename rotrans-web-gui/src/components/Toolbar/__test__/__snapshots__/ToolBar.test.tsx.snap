// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolBar component Matches with snapshot - without selected validator 1`] = `
Array [
  <div
    className="toolbar-details-row"
  >
    <div
      className="toolbar-title-div"
    >
      <span
        className="custom-header-title"
        data-testid="toolbarTitle"
      >
        Visualization
      </span>
      <span
        className="toolbar-sub-title"
      >
        Connection is established and secure.
      </span>
    </div>
    <div
      className="toolbar-profile-div"
    >
      <div
        className="toolbar-profile-component"
        onClick={[Function]}
      >
        <div
          className="toolbar-image-div"
        >
          <div
            className="toolbar-avatar"
          />
        </div>
        <div
          className="toolbar-name-div"
        >
          <span
            className="toolbar-name"
            data-testid="toolbarSelectedAgentName"
          >
            ABC
          </span>
          <span
            className="toolbar-ip"
            data-testid="toolbarSelectedAgentIp"
          >
            192.168.1.1
          </span>
        </div>
        <div
          className="toolbar-icon-div"
        >
          <img
            className="icon"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "down.svg",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>,
  <hr />,
]
`;

exports[`ToolBar component Matches with snapshot - without selected validator 2`] = `
Array [
  <div
    className="toolbar-details-row"
  >
    <div
      className="toolbar-title-div"
    >
      <span
        className="custom-header-title"
        data-testid="toolbarTitle"
      >
        Transactions
      </span>
      <span
        className="toolbar-sub-title"
      >
        Connection is established and secure.
      </span>
    </div>
    <div
      className="toolbar-profile-div"
    >
      <div
        className="toolbar-profile-component"
        onClick={[Function]}
      >
        <div
          className="toolbar-image-div"
        >
          <div
            className="toolbar-avatar"
          />
        </div>
        <div
          className="toolbar-name-div"
        >
          <span
            className="toolbar-name"
            data-testid="toolbarSelectedAgentName"
          >
            DEF
          </span>
          <span
            className="toolbar-ip"
            data-testid="toolbarSelectedAgentIp"
          >
            192.168.1.2
          </span>
        </div>
        <div
          className="toolbar-icon-div"
        >
          <img
            className="icon"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "down.svg",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>,
  <hr />,
]
`;
